index.d.ts

```diff
@@ -26,7 +27,22 @@ pupa('I like {{0}} and {{1}}', ['<br>🦄</br>', '<i>🐮</i>']);
 */
 declare function pupa(
 	template: string,
-	data: unknown[] | {[key: string]: any}
+	data: unknown[] | {[key: string]: any},
+	options?: {
+		/** By default, Pupa throws a `MissingValueError` when a placeholder resolves to `undefined`. With this option being `true`, it simply ignores it and leaves the placeholder as is. */
+		ignoreMissing?: boolean;
+		/** Performs arbitrary operation for each interpolation. If the returned value was `undefined`, it behaves different depending on `ignoreMissing` option. Otherwise, the returned value will be passed to `String` constructor (and escaped when double-braced) and embedded into the template. */
+		transform?: (data: {value: unknown; key: string}) => unknown;
+	}
```

Comment:
I prefer Options to be a separate type.	https://github.com/sindresorhus/typescript-definition-style-guide/tree/57a925bb5b3eb46a4666eab31e1c883a26b7bfb5#documentation

Also, follow the style there.

Reply:
Okay I'll make it separate, but sorry I don't have enough motivation to read through your style guide. Most people will think like me and your preference can't be enforced unless you provide some tools to automatically lint (that can be run by `yarn test`).

Reply:
No worries.

> enforced unless you provide some tools to automatically lint (that can be run by yarn test).

I do, but most of those enforcements require me to upgrade the linter which will require Node.js 12. And there are still many things that just cannot be automated.